company-ycmd.el

```diff
@@ -406,12 +406,12 @@ candidates list."
   (let* ((prefix-start-col (- (+ 1 (ycmd--column-in-bytes)) (length prefix)))
          (prefix-size (- start-col prefix-start-col))
          (prefix-diff (substring-no-properties prefix 0 prefix-size))
+         (prefix-diff-p (s-present? prefix-diff))
```

Comment:
That was probably just defensive coding on my part. The `set-text-property` call with indices 0 and 1 will throw an error on an empty string. I probably discovered that while I was getting this working and decided to make sure it could never happen. Is that check causing a performance problem?